app/javascript/components/patient/follow_up_flag/FollowUpFlagModal.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Modal } from 'react-bootstrap';
+
+import FollowUpFlag from './FollowUpFlag';
+
+class FollowUpFlagModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      initial_follow_up_reason: '',
+    };
+  }
+
+  componentDidMount() {
+    if (!this.props.bulk_action && this.props.patient.follow_up_reason) {
+      this.setState({ initial_follow_up_reason: this.props.patient.follow_up_reason });
+    }
+  }
+
+  render() {
+    return (
+      <Modal id="follow-up-flag-modal" size="lg" centered show={this.props.show} onHide={this.props.close}>
+        <Modal.Header closeButton>
+          <Modal.Title>
+            {!this.props.bulk_action && this.state.initial_follow_up_reason !== '' && !this.props.clear_flag && 'Update '}
```

Comment:
thoughts on leaving this Flag for Follow Up for update as well? or add a Set in front of the initial setting modal?  Also maybe just do Clear Flag on clear as well

Reply:
I like this a lot!!